### ISSUE TITLE
useRadio should set aria-checked prop input field to allow testing checked state

### ISSUE THREAD

**Issue by mauron85** [adobe/react-spectrum#5741]
### Provide a general summary of the issue here

Currently useRadio hook doesn't set aria-checked as inputProp for RadioComponent -> input

https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/RadioGroup.tsx#L169

So it's not possible to test this component using react testing library for checked state because of missing aria-checked on input element:

```
expect(
	screen.queryByRole('radio', {
		name: /International Single Trip/i
	})
).toBeChecked()
```

Form toBeChecked:

>Assert whether the given element is checked.
> It accepts an `input` of type `checkbox` or `radio` and elements with a `role` of `radio` with a valid
> `aria-checked` attribute of "true" or "false".

### 🤔 Expected Behavior?

Input element of Radio should have aria-checked attribute.

### 😯 Current Behavior

Input element of Radio doesn't have aria-checked attribute.

### 💁 Possible Solution

Add aria-checked as inputProps in useRadio hook

### 🖥️ Steps to Reproduce

Any example from:

https://react-spectrum.adobe.com/react-aria/RadioGroup.html

### Version

1.0.0

### What browsers are you seeing the problem on?

Microsoft Edge

### What operating system are you using?

Windows 10

**Comment by reidbarber** [comment]
This was removed in https://github.com/adobe/react-spectrum/pull/534 after an audit. Since the input has `type="radio"`, the native `checked` should suffice. I'm not sure why the jest matcher would fail, since the RTL docs list this as an example: https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobechecked. Maybe double check if the query is returning the correct element?

**Comment by mauron85** [comment]
@reidbarber yes, you are right. My test was actually broken. Closing this one as false alarm. Sorry for that.
